### PR TITLE
stash-cli 3.9.0 (new formula)

### PR DESCRIPTION
**Pull request by guoxiao**

### CODE REVIEW

File: Library/Formula/stash-cli.rb [hunk 1]
```diff
@@ -0,0 +1,17 @@
+class StashCli < Formula
+  homepage "https://marketplace.atlassian.com/plugins/org.swift.stash.cli"
+  url "https://bobswift.atlassian.net/wiki/download/attachments/16285777/stash-cli-3.9.0-distribution.zip?api=v2"
+  version "3.9.0"
+  sha1 "f615519894b194959754b9a7b5fb9bc03855dbcd"
+
+  def install
+    inreplace "stash.sh", "`dirname $0`", share
+    share.install "lib"
+    bin.install "./stash.sh"
```
Comment (mikemcquaid) [inline Library/Formula/stash-cli.rb]: `bin.install "stash.sh" => "stash"` which means you can remove the line below.

File: Library/Formula/stash-cli.rb [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+    File.rename prefix/"bin/stash.sh", prefix/"bin/stash"
+  end
+
+  test do
+    assert shell_output(bin/"stash --help 2>&1 | head").include?("Usage:")
```
Comment (mikemcquaid) [inline Library/Formula/stash-cli.rb]: Can this test be modified to do something more substantial than e.g. `--version` or `--help`? See `cmake.rb` for an example of an application formula with a good test and `tinyxml2.rb` for an example of a library formula with a good test. Thanks!

File: Library/Formula/stash-cli.rb (hunk 1 above)
Comment (guoxiao) [inline Library/Formula/stash-cli.rb]: aha that's what I wants. thanks.

File: Library/Formula/stash-cli.rb (hunk 2 above)
Comment (guoxiao) [inline Library/Formula/stash-cli.rb]: I tried using just `stash --help` here, but the returned value of it is not 0, which means the test was failed. `| head` is here only to change the returned value to 0. If there is some better solution please let me know. Thanks.

Comment (mikemcquaid) [inline Library/Formula/stash-cli.rb]: Is there any public repos (or BitBucket) that this tool can be run on to test?

File: Library/Formula/stash-cli.rb [hunk 3]
```diff
@@ -0,0 +1,16 @@
+class StashCli < Formula
+  homepage "https://marketplace.atlassian.com/plugins/org.swift.stash.cli"
+  url "https://bobswift.atlassian.net/wiki/download/attachments/16285777/stash-cli-3.9.0-distribution.zip?api=v2"
+  version "3.9.0"
+  sha1 "f615519894b194959754b9a7b5fb9bc03855dbcd"
+
+  def install
+    inreplace "stash.sh", "`dirname $0`", share
+    share.install "lib"
+    bin.install "stash.sh" => 'stash'
```
Comment (mikemcquaid) [inline Library/Formula/stash-cli.rb]: Use double-quotes: http://bot.brew.sh/job/Homebrew%20Pull%20Requests/19943/version=yosemite/testReport/junit/brew-test-bot/yosemite/audit_stash_cli___strict/

**Comment by mikemcquaid** [comment]
> A commercial license is required for version 3.0 or higher. See Atlassian CLI license for more information.
https://bobswift.atlassian.net/wiki/display/SCLI/Stash+Command+Line+Interface;jsessionid=3D77987CF99C4631CCEECFCFC80E036D

This makes me wonder if we should include this?

**Comment by guoxiao** [comment]
@mikemcquaid maybe I should submit this formula to `homebrew-binary` since it's not build from source?

**Comment by mikemcquaid** [comment]
Do you know what the licensing situation is?

**Comment by guoxiao** [comment]
I found the `LICENSE-CLI.txt` in the tarball which says:
> Copyright (c) 2005, 2012 Bob Swift and other contributors.
All rights reserved.
> Redistribution and use in source and binary forms, with or without
modification, are permitted provided that the following conditions are met:

>    * Redistributions of source code must retain the above copyright notice,
>      this list of conditions and the following disclaimer.
>    * Redistributions in binary form must reproduce the above copyright notice,
>      this list of conditions and the following disclaimer in the documentation
>      and/or other materials provided with the distribution.
>    * The names of contributors may not
>      be used to endorse or promote products derived from this software without
>      specific prior written permission.

As redistribution is allowed if we include the license, I don't think include this in homebrew is not appropriate.

**Comment by mikemcquaid** [comment]
Ok, cool thanks. Is there any public repos (or BitBucket) that this tool can be run on to test?

**Comment by guoxiao** [comment]
This tool can only run with git repositories hosted with stash, but I didn't find public demos on its homepage.

**Pull request closed without merging**